### ISSUE TITLE
Underline

### ISSUE THREAD

**Issue by kanilturgut** [windyzboy/CustomEditor#5]
I understand why you seperated underline span as AgsUnderlineSpan, because android underlines current text in edittext without underline toggle button clicked and it breaks your editor handle methods logic. 

But in this approach if you code something like customEditText.setTextHtml("<u>something</u>"), editor will not underline it, as Html.fromHtml (actually it is Html.withinParagraph) checks only UnderlineSpan to put <u> tags to text. 

With UnderlineSpan not AgsUnderline, some devices does not put automatic underline with some inputType filter, but other devices puts. For example, Nexus 4 (without root) works like charm, but S4 does not. 

If you find a way to solve this AgsUnderlineSpan / UnderlineSpan issue, please share with us.

**Comment by justintoth** [comment]
I have a similar issue in that when I'm retrieving the HTML from the customEditText (getTextHTML) the underline markup is not included in the returned HTML. Were you able to figure out a solution?

**Comment by kanilturgut** [comment]
@justintoth you should change AgsUnderlineSpan to UnderlineSpan. Because android Html class works with UnderlineSpan. Secondly, set customEditText inputType to "textPassword" as some keyboards tries to suggest word, and it breaks underline span.

**Comment by justintoth** [comment]
That worked perfectly, thanks! Is there any risk to using UnderlineSpan instead of AgsUnderlineSpan? (Why does AgsUnderlineSpan exist?)

**Comment by trietphm** [comment]
@justintoth @kanilturgut Hi guys, thank you for using my library.
It has been a long time since I wrote this for my small project. `AgsUnderlineSpan` is a customize `UnderlineSpan` supports for Japanese (and some other languages) which use Suggestion or Auto-correct to type. You can try with this input keyboard [Google Japanese Input](https://play.google.com/store/apps/details?id=com.google.android.inputmethod.japanese&hl=en)

And I don't work on Android any more, so I can't help you to solve this. So sorry.

You can take a look at this library, it may be better [RichEditor for Android](https://github.com/wasabeef/richeditor-android)

**Comment by kanilturgut** [comment]
@windyzboy actually I do thank you, as your library saved my life :) Btw you can close this issue.

**Comment by justintoth** [comment]
Yeah I do as well, I actually had to port this over to C# / Xamarin, which took a good day or so but it's working great.

**Comment by swamy2217** [comment]
Thanks, I removed AgsUnderlineSpan class and Changed AgsUnderlineSpan to UnderlineSpan in Customedittext Class.
